1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,102 @@
-function setAlarm() {}
+const body = document.querySelector('body');
+
+//for adding new pause button to the document
+const allButtonsEl = document.querySelector('.centre')
+const newPauseBtn = document.createElement('button');
+newPauseBtn.innerText = 'Pause';
+allButtonsEl.appendChild(newPauseBtn)
+
+// to set an id for the setinterval function that we are using below
+let intervalId;
+
+
+function setAlarm() {
+  
+  const timeRemainingEl = document.getElementById('timeRemaining');
+  const setTimerBtn = document.getElementById('set')
+  let inputTime = 0;
+
+  
+  
+  
+  setTimerBtn.addEventListener('click', () => {
+    let alarmSetInput = document.getElementById('alarmSet');
+    inputTime = alarmSetInput.value;
+
+    //call function every second. intervalId used to clear interval
+    intervalId = setInterval(() => {
+      controlTimer()
+    }, 1000)
+    
+    
+  });
+
+  //for pause button to stop counting down
+  newPauseBtn.addEventListener('click', () => {
+    clearInterval(intervalId);
+    body.style.background = 'white'
+  })
+
+
+  function controlTimer() {
+    console.log(inputTime.length)
+    if (inputTime < 10) {
+      timeRemainingEl.innerText = `Time Remaining: 00:0${inputTime}`; 
+    } else {
+      timeRemainingEl.innerText = 'Time Remaining: 00:' + inputTime;
+    }
+    
+    inputTime -= 1;
+    
+    if (inputTime <= 0) {
+      setInterval(() => {
+        let colors = ['red', 'blue'];
+        body.style.background = colors[Math.floor(Math.random(0, 3) * 2)] 
```

Comment:
This is really nice added extra (if slightly headache inducing when looking for too long!)